docs/sop/PROVISIONING_PRODUCTION.md

```diff
@@ -0,0 +1,490 @@
+
+```  
+AWS_PROFILE=Upstream terraform refresh  
+```  
+>**Note**: will associate the public IP address from the EIPs to the terraform variable for public IPs
+
+
+Next, Initialise Ansible including copying ansible files into `live` environment and save to Gitlab variables the IP addresses output by terraform with this command:
+```  
+../../../scripts/ansible_init.sh --env live  
+```  
+
+Install third party Ansible roles:
+```  
+ansible-galaxy install -r ../../../infrastructure/requirements.yml  
+```  
+
+Provision the bastion server:
+```  
+env OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES ansible-playbook -i ../../inventories bastion_playbook.yml --extra-vars="gigadb_env=live"  
+```  
+
+Provision web server:
+```  
+env TF_KEY_NAME=private_ip OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES ansible-playbook -i ../../inventories webapp_playbook.yml --extra-vars="gigadb_env=liveds
+```  
+
+Provision files server:
+```  
+env TF_KEY_NAME=private_ip OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES ansible-playbook -i ../../inventories files_playbook.yml --extra-vars="gigadb_env=live"  
+```
+
+
+
+### Provisioning live for upstream/alt-gigadb-website
+
+Change directory to the local checkout of the `upstream/alt-gigadb-website` project:
+
+```
+cd gigadb-alt-upstream
+```
+
+Change directory to the `envs` folder:
+```  
+cd ops/infrastructure/envs  
+```  
+
+Create directory `live` directory if not existing already and change to it:
+```  
+mkdir live
+```
+```  
+cd live  
+```  
+
+Initialise Terraform, including creating a new state (or retrieving if existing) on Gitlab and copying terraform files to `live` environment:
+```   
+../../../scripts/tf_init.sh --project gigascience/upstream/gigadb-website --env live  
+ ```
+It's an interactive command and it will ask question you will need to answer like:
+  ```
+	You need to specify the path to the ssh private key to use to connect to the EC2 instance: </path/to/your-ssh-key-for-sydney-region>  
+      
+    You need to specify your GitLab username: <user input>  
+      
+    You need to specify a backup file created by the files-url-updater tool: </path/to/giagdbv3_*_v9.3.5.backup> (optional)  
+      
+    You need to specify an AWS region: ap-southeast-2  
+ ```  
+
+Alternatively you can specify most of the requrested information as parameters to the commands:
+```  
+./../../scripts/tf_init.sh --project gigascience/upstream/gigadb-website --env live --region ap-southeast-2 --ssh-key /path/to/your-ssh-key-for-sydney-region --web-ec2-type t3.medium --bastion-ec2-type t3.medium --rds-ec2-type "t3.large"  
+```  
+
+You can now  provision production staging server using the appropriate profile:
+```  
+AWS_PROFILE=UpstreamAlt terraform show   
+```
+>**Note**: will show all the existing resources, which should be the same as the terraform state file `live_infra` from gitlab.
+
+```
+AWS_PROFILE=UpstreamAlt terraform plan
```

Comment:
**issue**: Executing this command resulted in errors:
```
│ Error: no matching EC2 EIP found
│ 
│   with module.ec2_dockerhost.data.aws_eip.docker_host_eip,
│   on ../../modules/aws-instance/aws-instance.tf line 114, in data "aws_eip" "docker_host_eip":
│  114: data "aws_eip" "docker_host_eip" {
│ 
╵
╷
│ Error: no matching EC2 EIP found
│ 
│   with module.files_host.data.aws_eip.docker_host_eip,
│   on ../../modules/aws-instance/aws-instance.tf line 114, in data "aws_eip" "docker_host_eip":
│  114: data "aws_eip" "docker_host_eip" {
│ 
╵
╷
│ Error: no matching EC2 EIP found
│ 
│   with module.ec2_bastion.data.aws_eip.bastion_host_eip,
│   on ../../modules/bastion-aws-instance/bastion-aws-instance.tf line 84, in data "aws_eip" "bastion_host_eip":
│   84: data "aws_eip" "bastion_host_eip" {
```
**suggestion**: I updated the live elastic IP addresses on AWS in Sydney region to resolve these errors.